nengo_gui/static/datastore.js

```diff
@@ -105,6 +98,7 @@ Nengo.DataStore.prototype.get_shown_data = function() {
 
     /* find the corresponding index values */
     var index = 0;
+    // Wouldn't a binary search be better?
```

Comment:
Probably, except if we make only a few loops until the value is found (I don't know this part of the code enough to judge how many loops are expected on average.)
Also depends on how often this code is called (if it's only called a few times per execution, it might not be worth to improve the efficiency, but this seems to be called on a regular basis?)

How regular are the values in `this.times`? If we expect a constant increase in most cases (with maybe a few dropped times), we could use a heuristic to estimate the index which might be even better than binary search.